### ISSUE TITLE
.psc-package/local/.set/packages.json does not exist

### ISSUE THREAD

**Issue by shmish111** [justinwoo/psc-package2nix#47]
I'm using `pp2n` to generate a packages.nix file and then in my main derivation I'm "installing" everything into `.psc-package` using the same technique as you use in your shell derivations. However there is no packages.json around. I suppose some tool should be able to generate one from packages.nix but I couldn't see anything that does this?

**Comment by justinwoo** [comment]
Sorry, I think this is caused by an older version of this. Are you are the latest commit? Or did you pull it down from easy-purescript-nix at some other hash?

**Comment by justinwoo** [comment]
If you run pp2n psc-package2nix, that should be available from a nix build, but this should not be necessary for running pp2n build or pp2n install

**Comment by justinwoo** [comment]
Sorry, maybe I just misunderstood what you were trying to get done. Are you starting with a Spago project or something? You might take a look at the setup I have in my vidtracker project: https://github.com/justinwoo/vidtracker/pull/28/files

Such that to generate a packages.nix, you run

```
	spago psc-package-insdhall
	pp2n psc-package2nix
```

then to install, you should not need to run the spago psc-package-install again but simply `pp2n install` (with an optional `-j 100` flag to tell nix to use more jobs)

**Comment by shmish111** [comment]
so I ran `spago psc-package-insdhall && pp2n psc-package2nix` in order to generate a nix file I can use in CI. I'm not sure what we will use in development for the moment but we do want to use dhall managed package sets however in CI we need to build using webpack. This can use psc-package so I need to get everything in the right place. I can now get all the packages in the right place in my derivation, but I just don't have a package.json file for the package set.

I just tried to generate packages.nix with 
```
      pp2n = import (pkgs.fetchFromGitHub {
        owner = "justinwoo";
        repo = "psc-package2nix";
        rev = "cc48ccd64862366a44b4185a79de321f93755782";
        sha256 = "0cvd1v3d376jiwh4rfhlyijxw3j6jp9rkm9hdb7k7sjxqs1dsviv";
      }) {};
```
and I still have the same issue, the packages.nix derivation doesn't say anything about how to generate what is expected in `.psc-package/local/.set/packages.json` so the purescript webpack plugin fails because it can't find this file (presumably it's actually `psc-package` failing.

**Comment by shmish111** [comment]
Just to be a bit clearer for context. We use webpack because it enables us to run a hot reloading server with packaged up sass and purescript, with hashed names so caching works correctly. We are happy to move to a different solution but I don't want to do that right now, I think it shouldn't be relevant. At the end of the day it uses psc-package and what I want/need is to use psc-package in a (pure) derivation. `spago psc-package-insdhall && pp2n psc-package2nix` almost gives me that, I just build the derivations and link to them in my derivation, just missing the final packages.json file to work now I think. I need some way to generate that from the packages.nix file, I thought you might have already done this part?

**Comment by justinwoo** [comment]
Could you update that rev? It's over two months old now.

```
{
  owner = "justinwoo";
  repo = "psc-package2nix";
  rev = "6e8f6dc6dea896c71b30cc88a2d95d6d1e48a6f0";
  sha256 = "0fa6zaxxmqxva1xmnap9ng7b90zr9a55x1l5xk8igdw2nldqfa46";
}
```

**Comment by shmish111** [comment]
OK, I have got past my confusion about what was needed and what was going on. I had to use the `.psc-package/local/.set/packages.json` file generated by `spago psc-package-insdhall` in my derivation. In addition I need:
```
  pp2n = pkgs.fetchFromGitHub {
    owner = "justinwoo";
    repo = "psc-package2nix";
    rev = "cc48ccd64862366a44b4185a79de321f93755782";
    sha256 = "0cvd1v3d376jiwh4rfhlyijxw3j6jp9rkm9hdb7k7sjxqs1dsviv";
  };
  mkCopyHook = import "${pp2n}/nix/mkCopyHook.nix";
  installPackages = builtins.toString (builtins.map (mkCopyHook packages) (builtins.attrValues packages.inputs));
```
With these 2 pieces I can `psc-package install` in my configure phase. Now anything that uses psc-package in the build phase should work (in my case webpack).

I will close this issue as it is not an issue however things are all over the place, I feel like there must be a nice way to glue all this together so I may open up some suggestions once I've cleaned up all my stuff and thought about it some more.

Thanks for your help @justinwoo and sorry for the noise!
